pkg/controller/bundle_sync_task.go

```diff
@@ -281,100 +282,106 @@ func (st *bundleSyncTask) handleProcessResult(retriable bool, processErr error)
 		return false, processErr
 	}
 
-	// Construct resource conditions and check if there were any resource errors
-	resourceStatuses := make([]smith_v1.ResourceStatus, 0, len(st.processedResources))
-	var failedResources []smith_v1.ResourceName
-	retriableResourceErr := true
 	statusUpdated := false
-	for _, res := range st.bundle.Spec.Resources { // Deterministic iteration order
-		blockedCond := smith_v1.ResourceCondition{Type: smith_v1.ResourceBlocked, Status: smith_v1.ConditionFalse}
-		inProgressCond := smith_v1.ResourceCondition{Type: smith_v1.ResourceInProgress, Status: smith_v1.ConditionFalse}
-		readyCond := smith_v1.ResourceCondition{Type: smith_v1.ResourceReady, Status: smith_v1.ConditionFalse}
-		errorCond := smith_v1.ResourceCondition{Type: smith_v1.ResourceError, Status: smith_v1.ConditionFalse}
-
-		if resInfo, ok := st.processedResources[res.Name]; ok {
-			// Resource was processed
-			switch resStatus := resInfo.status.(type) {
-			case resourceStatusDependenciesNotReady:
-				blockedCond.Status = smith_v1.ConditionTrue
-				blockedCond.Reason = smith_v1.ResourceReasonDependenciesNotReady
-				blockedCond.Message = fmt.Sprintf("Not ready: %q", resStatus.dependencies)
-			case resourceStatusInProgress:
-				inProgressCond.Status = smith_v1.ConditionTrue
-			case resourceStatusReady:
-				readyCond.Status = smith_v1.ConditionTrue
-			case resourceStatusError:
-				errorCond.Status = smith_v1.ConditionTrue
-				errorCond.Message = resStatus.err.Error()
-				if resStatus.isRetriableError {
-					errorCond.Reason = smith_v1.ResourceReasonRetriableError
+	// TODO: We might want to update the status even if bundle is marked for
+	// deletion (e.g. report which resources have been deleted and which block
+	// the deletion of the bundle), but currently it will generate useless "Unknown"
```

Comment:
Marking the status as deletion in progress would be useful for reporting tooling. (even if we don't list the resources that could be blocking deletion)